addons/purchase_stock/models/purchase.py

```diff
@@ -312,16 +297,41 @@ def _create_or_update_picking(self):
                     line.invoice_lines[0].move_id.activity_schedule(
                         'mail.mail_activity_data_warning',
                         note=_('The quantities on your purchase order indicate less than billed. You should ask for a refund.'))
-
-                # If the user increased quantity of existing line or created a new line
-                pickings = line.order_id.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel') and x.location_dest_id.usage in ('internal', 'transit'))
-                picking = pickings and pickings[0] or False
-                if not picking:
-                    res = line.order_id._prepare_picking()
-                    picking = self.env['stock.picking'].create(res)
-
-                moves = line._create_stock_moves(picking)
-                moves._action_confirm()._action_assign()
+                if float_compare(previous_product_qty[line], line.product_qty, line.product_uom.rounding) >= 0:
+                    if not float_is_zero(line.qty_received, line.product_uom.rounding)\
+                            and float_compare(line.product_qty, line.qty_received, line.product_uom.rounding) > 0:
+                        # is there any quantity to decrease (in a backorder)
+                        open_moves = line.move_ids.filtered(lambda move: move.state not in ('done', 'cancel') and move.location_dest_id.usage != "supplier")
+                        if open_moves and sum(open_moves.mapped('product_uom_qty')) > (line.product_qty - line.qty_received):
+                            line.move_ids._decrease_initial_demand(previous_product_qty[line] - line.product_qty, 'DOWN')
+                        else:
+                            picking = line.order_id.picking_ids.filtered(
+                                lambda x: x.state not in ('done', 'cancel') and x.location_dest_id.usage in ('internal', 'transit')
+                            )
+                            if not picking:
+                                picking = self.env['stock.picking'].create(
+                                    line.order_id._prepare_picking()
+                                )
+                            price_unit = self._get_stock_move_price_unit()
+                            product_uom_qty, product_uom = self.product_uom._adjust_uom_quantities(
+                                line.product_qty - line.qty_received, self.product_id.uom_id
+                            )
+                            new_moves |= self.env['stock.move'].create(
+                                self._prepare_stock_move_vals(picking, price_unit, product_uom_qty, product_uom)
+                            )
+                    else:
+                        line.move_ids._decrease_initial_demand(previous_product_qty[line] - line.product_qty, 'DOWN')
```

Comment:
move_dest_ids if you still in an RFQ state?

Reply:
_create_or_update_picking() is called on state = 'purchase' lines only